lib/index.ts

```diff
@@ -1,172 +1,38 @@
-import * as Debug from "debug";
-import * as path from "path";
-
-import * as analyzer from "./analyzer";
-import { buildTree } from "./dependency-tree";
-import { Docker, DockerOptions } from "./docker";
 import * as dockerFile from "./docker-file";
-import { tryGetAnalysisError } from "./errors";
 import { experimentalAnalysis } from "./experimental";
-import { getRuntime } from "./inputs/runtime/docker";
-import { parseAnalysisResults } from "./parser";
-import { buildResponse } from "./response-builder";
-import * as staticUtil from "./static";
 import {
-  DepTree,
-  ManifestFile,
+  ContainerTarget,
+  Fact,
+  Identity,
+  PluginOptions,
   PluginResponse,
-  ScannedProjectExtended,
-  ScannedProjectManifestFiles,
-  ScanType,
+  ScanResult,
 } from "./types";
 
 export {
-  inspect,
+  scan,
   dockerFile,
+  ScanResult,
   PluginResponse,
-  ScannedProjectExtended,
-  ScanType,
-  ScannedProjectManifestFiles,
-  DepTree,
+  ContainerTarget,
+  Identity,
+  Fact,
 };
 
-const MAX_MANIFEST_FILES = 5;
-const debug = Debug("snyk");
-
-async function inspect(
-  root: string,
-  targetFile?: string,
-  options?: any,
-): Promise<PluginResponse> {
-  const targetImage = root;
-
-  const dockerfileAnalysis = await dockerFile.readDockerfileAndAnalyse(
-    targetFile,
-  );
-
-  if (options && options.experimental) {
-    return await experimentalAnalysis(targetImage, dockerfileAnalysis, options);
+async function scan(options?: Partial<PluginOptions>): Promise<PluginResponse> {
+  if (!options) {
+    throw new Error("No plugin options provided");
   }
 
-  if (staticUtil.isRequestingStaticAnalysis(options)) {
-    return await staticUtil.analyzeStatically(
-      targetImage,
-      dockerfileAnalysis,
-      options,
-    );
+  const targetImage = options.path;
```

Comment:
what is this `path` parameter? it's the first time I see it

Reply:
This used to be `root` - the first parameter passed to the plugin's `inspect`. It was always referred to as `path` in the CLI, I think it is because of the assumption the CLI is scanning some filesystem path.